javascript/javascript2/week3/theFastestPresser/theFasterPresser.js

```diff
@@ -0,0 +1,93 @@
+
+ 
+ 
+ const input= document.getElementById('input'); 
+ const button=document.getElementById('game'); 
+ const dCounter=document.getElementById('count-d'); 
+ const kCounter=document.getElementById('count-k'); 
+
+//  document.getElementById('count-d').innerHTML=totald; 
+//  document.getElementById('count-k').innerHTML= totalk; 
+ let total=0;
+ let totald=0; 
+ let totalk=0; 
+ button.addEventListener('click', handleClick); 
+
+ function handleClick( ){
+  const time= input.value; 
+  document.addEventListener('keyup', handlekeypress); 
+
+     setTimeout( endGame, time*1000)
+ }; 
+
+
+function endGame(){
+    document.removeEventListener('keyup', handlekeypress)
+};
+endGame(); 
+
+
+ //Create an event listener so you can call a function when any key is pressed.
+ // Now grap the actual key that was pressed
+// document.addEventListener('keypress', handleKeyPress); 
+  
+//  const h2= document.getElementsByTagName('h2'); 
+//  document.addEventListener('keypress', handlekeypress)
+//  document.getElementById('pressD"')
+
+function handlekeypress(press){
+    if (input.value === 0) {
+        press.preventDefault()
+        return
+    }
+if(press.keyCode === 68){
+        totald++; 
+     console.log(totald);
+    dCounter.innerHTML = totald; 
+    
+
+}
+
+   else if( press.keyCode === 75){
+        totalk++; 
+        console.log(totalk);
+kCounter.innerHTML= totalk;
+ 
+}
+};
+
+function winner(){
+    if( totald > totalk){
+         alert(' The winner is D');
+        confetti.render();
+    }
+    else if(  totald < totalk ){
+      alert( ' The winner is K');
+      confetti.render(); 
+    } 
+    else {
+        alert('no winner in the Game!');
+    }
+}
+  winner(); 
```

Comment:
You should move this into endGame() so your winner function is called after the game is finished.